chromadb/db/impl/sqlite.py

```diff
@@ -246,3 +257,39 @@ def uuid_to_db(uuid: Optional[UUID]) -> Optional[Any]:
     @override
     def unique_constraint_error() -> Type[BaseException]:
         return sqlite3.IntegrityError
+
+    @trace_method("SqliteDB.optimize_system", OpenTelemetryGranularity.ALL)
+    @override
+    def optimize_system(self) -> SystemOptimizationStats:
+        super().optimize_system()
+        # get size of DB file
+        if not self._is_persistent:
+            # we don't need to optimize an in-memory DB
+            return SystemOptimizationStats(
+                db_size_before=-1,
+                db_size_after=-1,
+            )
+        file_size_before = os.path.getsize(self._db_file)
+        self._conn_pool.close()
+        _optimization_lock.acquire()
+        try:
+            conn = sqlite3.connect(self._db_file)
+            conn.execute("PRAGMA busy_timeout = 300000;")  # Wait up to 300 seconds
```

Comment:
@HammadB, we can move all of the below in an external script. It will make it a bit more accessible to extend.